core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -86,16 +85,15 @@ protected CachingCatalog(Catalog catalog, boolean caseSensitive, long expiration
   }
 
   /**
-   * CacheWriter class for removing metadata tables when their associated data table is expired
+   * RemovalListener class for removing metadata tables when their associated data table is expired
    * via cache expiration.
+   *
+   * @see com.github.benmanes.caffeine.cache.RemovalListener
    */
-  class MetadataTableInvalidatingCacheWriter implements CacheWriter<TableIdentifier, Table> {
+  class MetadataTableInvalidatingRemovalListener implements RemovalListener<TableIdentifier, Table> {
     @Override
-    public void write(TableIdentifier tableIdentifier, Table table) {
-    }
-
-    @Override
-    public void delete(TableIdentifier tableIdentifier, Table table, RemovalCause cause) {
+    public void onRemoval(TableIdentifier tableIdentifier, Table table, RemovalCause cause) {
+      identLoggingRemovalListener.onRemoval(tableIdentifier, table, cause);
```

Comment:
Nit: Instead of relying on `identLoggingRemovalListener`, since we're already inside of an `onRemoval` function, would it make sense to cut out the overhead and just add the log statement here if need be?

The double `onRemoval` call makes me nervous about more potential race conditions etc by looking like a double callback.

Reply:
removed `identLoggingRemovalListener`